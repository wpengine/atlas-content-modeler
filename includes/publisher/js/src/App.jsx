/* global atlasContentModelerFormEditingExperience */
import React, { useEffect, useState } from "react";
import Fields from "./components/Fields";
import { __ } from "@wordpress/i18n";
import { sendEvent } from "acm-analytics";
import TrashPostModal from "./components/TrashPostModal";

export default function App({ model, mode }) {
	const isEditMode = mode === "edit";
	const [trashPostModalIsOpen, setTrashPostModalIsOpen] = useState(false);

	useEffect(() => {
		const queryParams = new URLSearchParams(window.location.search);
		const newPost = queryParams.get("acm-post-published");
		if (newPost) {
			sendEvent({
				category: "post",
				action: "Post Published",
			});
		}
	}, []);

	return (
		<div className="app classic-form" style={{ marginTop: "20px" }}>
			<div className="flex-parent">
				<div>
					<h3 className="main-title">
						{isEditMode ? (
							<span>{__("Edit", "atlas-content-modeler")} </span>
						) : (
							<span>{__("Add", "atlas-content-modeler")} </span>
						)}
						{model.singular}
					</h3>
				</div>

				{isEditMode && (
					<div
						style={{ marginLeft: "20px" }}
						className="flex-align-v"
					>
						<a
							className="page-title-action"
							href={
								"/wp-admin/post-new.php?post_type=" + model.slug
							}
						>
							Add New
						</a>
					</div>
				)}
			</div>
			<div className="d-flex flex-column">
				<Fields model={model} />
			</div>
			<TrashPostModal
				isOpen={trashPostModalIsOpen}
				setIsOpen={setTrashPostModalIsOpen}
			/>
		</div>
	);
}
