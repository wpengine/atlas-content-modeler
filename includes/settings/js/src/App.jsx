import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { ToastContainer, Flip } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

import CreateContentModel from "./components/CreateContentModel.jsx";
import ViewContentModelsList from "./components/ViewContentModelsList";
import EditContentModel from "./components/EditContentModel";
import Taxonomies from "./components/Taxonomies";
import Tools from "./components/Tools";
import Settings from "./components/Settings";
import { useLocationSearch } from "./utils";
import { ModelsContextProvider } from "./ModelsContext";

export default function App() {
	return (
		<div className="app atlas-content-modeler">
			<ModelsContextProvider>
				<Router>
					<ToastContainer
						autoClose={2000}
						draggable={false}
						position="top-center"
						transition={Flip}
					/>
					<ViewTemplate />
				</Router>
			</ModelsContextProvider>
		</div>
	);
}

function ViewTemplate() {
	const query = useLocationSearch();
	const view = query.get("view");

	if (view === "create-model") {
		return <CreateContentModel />;
	}

	if (view === "edit-model") {
		return <EditContentModel />;
	}

	if (view === "taxonomies") {
		return <Taxonomies />;
	}

	if (view === "tools") {
		return <Tools />;
	}

	if (view === "settings") {
		return <Settings />;
	}

	return <ViewContentModelsList />;
}
