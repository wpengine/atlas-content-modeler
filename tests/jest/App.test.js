import React from "react";
import renderer from "react-test-renderer";
import App from "../../includes/settings/js/src/App";

describe("App", () => {
	const app = renderer.create(<App />);

	let tree = app.toJSON();

	it("Renders a matching snapshot", () => {
		expect(tree).toMatchSnapshot();
	});
});
