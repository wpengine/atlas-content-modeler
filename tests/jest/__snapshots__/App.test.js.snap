// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders a matching snapshot 1`] = `
<div
  className="app atlas-content-modeler"
>
  <div
    className="Toastify"
  />
  <div
    className="css-1y92ofa-Card"
  >
    <section
      className="flex-wrap d-flex flex-column d-sm-flex flex-sm-row css-1pjtado-Header"
    >
      <h2>
        Content Models
      </h2>
    </section>
    <section
      className="card-content"
    >
      <div
        className="model-list-empty"
      >
        <p>
          You currently have no Content Models.
        </p>
        <button
          className="css-1cfhljd-Button"
          onClick={[Function]}
        >
          Get Started
        </button>
      </div>
    </section>
  </div>
</div>
`;
